src/Microsoft.AspNet.FileProviders.Combined/CombinedFileProvider.cs

```diff
@@ -0,0 +1,168 @@
+
+            // There is no existing directory contents
+            if (existingDirectoryContentsForAllProviders.Count == 0)
+            {
+                return new NotFoundDirectoryContents();
+            }
+            var combinedDirectoryContents = new CombinedDirectoryContents(existingDirectoryContentsForAllProviders);
+            return combinedDirectoryContents;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="IChangeToken"/> for the specified <paramref name="filter"/>.
+        /// </summary>
+        /// <remarks></remarks>
+        /// <param name="pattern">Filter string used to determine what files or folders to monitor. Example: **/*.cs, *.*, subFolder/**/*.cshtml.</param>
+        /// <returns>An <see cref="IChangeToken"/> that is notified when a file matching <paramref name="filter"/> is added, modified or deleted. The change token will be notified when one of the change token returned by the provided <see cref="IFileProvider"/> will be notified.</returns>
+        public IChangeToken Watch(string pattern)
+        {
+            // Watch all file providers
+            var activeChangeTokens = _fileProviders
+                .Select(fileProvider => fileProvider.Watch(pattern))
+                .Where(changeToken => changeToken != null && changeToken.ActiveChangeCallbacks).ToList();
+
+            // There is no change token with active change callbacks
+            if (activeChangeTokens.Count == 0)
+            {
+                return NoopChangeToken.Singleton;
+            }
+            var combinedFileChangeToken = new CombinedFileChangeToken(activeChangeTokens);
+            return combinedFileChangeToken;
+        }
+
+        private class CombinedDirectoryContents : IDirectoryContents
+        {
+            private readonly Dictionary<string, IFileInfo> _files = new Dictionary<string, IFileInfo>();
+
+            public CombinedDirectoryContents(IEnumerable<IEnumerable<IFileInfo>> listOfFiles)
+            {
+                foreach (var files in listOfFiles)
+                {
+                    Exists = true;
+                    foreach (var file in files)
+                    {
+                        if (!_files.ContainsKey(file.Name))
+                        {
+                            _files.Add(file.Name, file);
+                        }
+                    }
+                }
+            }
+
+            public IEnumerator<IFileInfo> GetEnumerator()
+            {
+                return _files.Values.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return _files.Values.GetEnumerator();
+            }
+
+            public bool Exists { get; }
+        }
+
+        private class CombinedFileChangeToken : IChangeToken
+        {
+            private readonly IEnumerable<IChangeToken> _changeTokens;
+
+            public CombinedFileChangeToken(IEnumerable<IChangeToken> changeTokens)
```

Comment:
Same remark as above: since this parameter is always used with a list, you could replace IEnumerable with a list or an array.